### ISSUE TITLE
Better organized output_test

### ISSUE THREAD

**Issue by yongruilin** [jpbetz/validation-gen#10]
### What would you like to be added?

Update the tests structure under  [output_tests](https://github.com/jpbetz/kubernetes/tree/a2e6b49acbf40c7017870ddc16a625759ef941b8/staging/src/k8s.io/code-generator/cmd/validation-gen/output_tests). 2 types of testing, one for testing the specific validation, the other for testing whether the validation tags are applied as expected.

```
output_tests
├── enum
├── forbidden
├── maximum
├── maxitems
├── maxlength
├── optional
├── required
...
...
├── validation_true
└── validation_false
```
The `enum`, `forbidden`, `maxitems` etc folders hold the tests for validation logic of tag `+enum`, `+forbidden`, `+maxitems` etc. And the `validation_true` and `validation_false` folders hold the tests for whether the tags are applied as expected.

### Why is this needed?

The tests under [output_tests](https://github.com/jpbetz/kubernetes/tree/a2e6b49acbf40c7017870ddc16a625759ef941b8/staging/src/k8s.io/code-generator/cmd/validation-gen/output_tests) are un-organized well and hard to follow the tests pattern.

**Comment by thockin** [comment]
We moved all the tag-related tests to a tags dir - do we need more here?
